tests/e2e/scale.go

```diff
@@ -0,0 +1,139 @@
+
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+		//Delete storage class if already present
+		sc, err := client.StorageV1().StorageClasses().Get(ctx, storagePolicyName, metav1.GetOptions{})
+		if err == nil && sc != nil {
+			gomega.Expect(client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))).NotTo(gomega.HaveOccurred())
+		}
+
+		if guestCluster {
+			svcClient, svNamespace := getSvcClientAndNamespace()
+			setResourceQuota(svcClient, svNamespace, rqLimitScaleTest)
+		}
+
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		if supervisorCluster {
+			deleteResourceQuota(client, namespace)
+		}
+
+		for _, pvc := range pvclaims {
+			pvclaimToDelete, err := client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
+			if err == nil {
+				err := fpv.DeletePersistentVolumeClaim(client, pvclaimToDelete.Name, pvclaimToDelete.Namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+
+		}
+
+	})
+
+	/*
+		Test to Create and Delete PVC
+		Steps
+			1.	Create a Storage Class
+			2.	Create PVC using above SC (pre data)
+			3.	Create and Delete PVC (NUMBER_OF_GO_ROUTINES * WORKER_PER_ROUTINE) times
+			4.	Delete SC
+	*/
+	ginkgo.It("Scale PVC", func() {
+		var storageclass *storagev1.StorageClass
+		var err error
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ginkgo.By("Invoking Test to create and delete PVCs")
+		// decide which test setup is available to run
+		if supervisorCluster {
+			ginkgo.By("CNS_TEST: Running for WCP setup")
+			profileID := e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+			scParameters[scParamStoragePolicyID] = profileID
+			// create resource quota
+			createResourceQuota(client, namespace, rqLimitScaleTest, storagePolicyName)
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		storageclass, err = createStorageClass(client, scParameters, nil, "", "", true, storagePolicyName)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Create Storage class and PVC
+		ginkgo.By("Creating Storage Class and PVC with allowVolumeExpansion = true")
+		var wg sync.WaitGroup
+		lock := &sync.Mutex{}
+		goRoutine := GetAndExpectIntEnvVar(envNumberOfGoRoutines)
```

Comment:
Update md file if these params are mandatory.
Otherwise use a default value for each of the params, if not present or skip the test.

Reply:
I have modified the code to skip the tests if the environment params are not passed.